httptransport/updatediffhandler.go

```diff
@@ -0,0 +1,131 @@
+package httptransport
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/google/uuid"
+	"github.com/quay/claircore/pkg/jsonerr"
+
+	"github.com/quay/clair/v4/matcher"
+)
+
+type updateDiffHandler struct {
+	matcher.Differ
+}
+
+func (h *updateDiffHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
When I look at this I am not sure what your handler is doing at first glance. IMO it's much better to have a simple function which directly correlates with what's happening in the handler, not a handler thats doing some muxing itself.